internal/service/privatelinkendpointserverless/resource_privatelink_endpoint_serverless.go

```diff
@@ -133,41 +128,38 @@ func resourceMongoDBAtlasPrivateLinkEndpointServerlessRead(ctx context.Context,
 		return diag.Errorf("error getting Serverless private link endpoint  information: %s", err)
 	}
 
-	if err := d.Set("endpoint_id", privateLinkResponse.ID); err != nil {
+	if err := d.Set("endpoint_id", privateLinkResponse.GetId()); err != nil {
 		return diag.Errorf("error setting `endpoint_id` for endpoint_id (%s): %s", d.Id(), err)
 	}
 
 	if err := d.Set("instance_name", instanceName); err != nil {
 		return diag.Errorf("error setting `instance Name` for endpoint_id (%s): %s", d.Id(), err)
 	}
 
-	if err := d.Set("endpoint_service_name", privateLinkResponse.EndpointServiceName); err != nil {
+	if err := d.Set("endpoint_service_name", privateLinkResponse.GetEndpointServiceName()); err != nil {
 		return diag.Errorf("error setting `endpoint_service_name Name` for endpoint_id (%s): %s", d.Id(), err)
 	}
 
-	if err := d.Set("private_link_service_resource_id", privateLinkResponse.PrivateLinkServiceResourceID); err != nil {
+	if err := d.Set("private_link_service_resource_id", privateLinkResponse.GetPrivateLinkServiceResourceId()); err != nil {
 		return diag.Errorf("error setting `private_link_service_resource_id Name` for endpoint_id (%s): %s", d.Id(), err)
 	}
 
-	if err := d.Set("status", privateLinkResponse.Status); err != nil {
+	if err := d.Set("status", privateLinkResponse.GetStatus()); err != nil {
 		return diag.FromErr(fmt.Errorf(privatelinkendpoint.ErrorPrivateLinkEndpointsSetting, "status", d.Id(), err))
 	}
 
 	return nil
 }
 
-func resourceMongoDBAtlasPrivateLinkEndpointServerlessDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	// Get client connection.
-	conn := meta.(*config.MongoDBClient).Atlas
+func resourceDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	ids := conversion.DecodeStateID(d.Id())
 	projectID := ids["project_id"]
 	instanceName := ids["instance_name"]
 	endpointID := ids["endpoint_id"]
 
-	_, _, err := conn.ServerlessPrivateEndpoints.Get(ctx, projectID, instanceName, endpointID)
+	_, _, err := connV2.ServerlessPrivateEndpointsApi.GetServerlessPrivateEndpoint(ctx, projectID, instanceName, endpointID).Execute()
 	if err != nil {
-		// case 404
-		// deleted in the backend case
 		if strings.Contains(err.Error(), "404") || strings.Contains(err.Error(), "400") {
```

Comment:
I think this comment in particular might be useful because it explains WHY not just mimic what the code does. in this case i keep it but in a line, like: // case 404: deleted in the backend case

Reply:
I think the HTTP code make it quite clear what is going on so I removed the comment. Within the context of delete operation, if resource is not found it is my understanding that the resource has been already deleted.

Reply:
I mean about the fact that the resource might be deleted in the backend and we want to handle it differently.  Without that context i would wonder why we want to do a special case here

Reply:
Okay, will add the comment again in both places.